File: Uni/Semester 1 - 2020/SENG2200 - Programming Languages and Paradigms/Assignments/Assignment1/Point.java
package assignment1;

public class Point
{
  private float x;
  private float y;

  public Point(float _x, float _y)
  {
    x = _x;
    y = _y;
  }

  public float GetX()
  {
    return x;
  }

  public float GetY()
  {
    return y;
  }

  public String ToString()
  {
    return "(" + Double.ToString(x) + " , " + Double.ToString(y) + ")";
    // return "(" + String.format("%d", x) + " , " + String.format("%d", y) + ")";
  }
}

File: Uni/Semester 1 - 2018/SENG1120 - Data Structures/Workshops/Week 3/Point.cpp
#include <cmath>
#include <iostream>
#include "Point.h"

using namespace std;

Point::Point()
{
	x = 0.0;
	y = 0.0;
}

Point::Point(double newX, double newY)
{
	x = newX;
	y = newY;
}

void Point::set_x(double newX)
{
	x = newX;
}

void Point::set_y(double newY)
{
	y = newY;
}

double Point::get_x() const
{
	return x;
}

double Point::get_y() const
{
	return y;
}

ostream& operator << (ostream& osObject, const Point& myPoint)
{
	osObject << "(" << myPoint.get_x() << "," << myPoint.get_y() << ")";
	return osObject;
}
